src/freenet/node/Node.java

```diff
@@ -2536,7 +2542,30 @@ public void set(Boolean val) throws InvalidConfigValueException,
 			
 		});
 		enableRoutedPing = nodeConfig.getBoolean("enableRoutedPing");
-		
+
+		nodeConfig.register("enableNodeDiagnostics", false, sortOrder++, true, false, "Node.enableDiagnostics", "Node.enableDiagnosticsLong", new BooleanCallback() {
+			@Override
+			public Boolean get() {
+				synchronized(Node.this) {
+					return enableNodeDiagnostics;
+				}
+			}
+
+			@Override
+			public void set(Boolean val) throws InvalidConfigValueException,
+				NodeNeedRestartException {
+				synchronized(Node.this) {
```

Comment:
Minor formatting nitpicking: as `synchronized` is a keyword there should be a space between it and the opening parens.

Reply:
Did a bit of clean up here. Let me know if there's a formatting issue.